ambry-server/src/main/java/com/github/ambry/server/mysql/AccountStatsMySqlStoreFactory.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server.mysql;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.config.AccountStatsMySqlConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.StatsManagerConfig;
+import com.github.ambry.config.VerifiableProperties;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.mysql.MySqlUtils.*;
+
+
+/**
+ * Factory to create a {@link AccountStatsMySqlStore}.
+ */
+public class AccountStatsMySqlStoreFactory {
+  private static final Logger logger = LoggerFactory.getLogger(AccountStatsMySqlStoreFactory.class);
+
+  private final AccountStatsMySqlConfig accountStatsMySqlConfig;
+  private final String localDC;
+  private final String clustername;
+  private final String hostname;
+  private final MetricRegistry registry;
+  private final String localBackupFilePath;
+
+  /**
+   * Constructor to create a {@link AccountStatsMySqlStoreFactory}.
+   * @param verifiableProperties
+   * @param clusterMapConfig
+   * @param statsManagerConfig
+   * @param registry
+   */
+  public AccountStatsMySqlStoreFactory(VerifiableProperties verifiableProperties, ClusterMapConfig clusterMapConfig,
+      StatsManagerConfig statsManagerConfig, MetricRegistry registry) {
+    accountStatsMySqlConfig = new AccountStatsMySqlConfig(verifiableProperties);
+    clustername = clusterMapConfig.clusterMapClusterName;
+    String[] domainNamesToRemove = accountStatsMySqlConfig.domainNamesToRemove.split(",");
+
+    String fullyQualifiedDomainName = clusterMapConfig.clusterMapHostName;
+    int port = clusterMapConfig.clusterMapPort;
+    for (String domainName : domainNamesToRemove) {
+      if (domainName.charAt(0) != '.') {
+        domainName = "." + domainName;
+      }
+      fullyQualifiedDomainName = fullyQualifiedDomainName.replace(domainName, "");
+    }
+    hostname = String.format("%s_%d", fullyQualifiedDomainName, port);
+    localDC = clusterMapConfig.clusterMapDatacenterName;
+    localBackupFilePath = statsManagerConfig.outputFilePath;
+    this.registry = registry;
+  }
+
+  /**
+   * Return {@link AccountStatsMySqlStore}.
+   * @return
+   * @throws SQLException
+   */
+  public AccountStatsMySqlStore getAccountStatsMySqlStore() throws SQLException {
+    Map<String, List<DbEndpoint>> dcToMySqlDBEndpoints = getDbEndpointsPerDC(accountStatsMySqlConfig.dbInfo);
+    // Flatten to List (TODO: does utility method need to return map?)
+    List<DbEndpoint> dbEndpoints = new ArrayList<>();
+    dcToMySqlDBEndpoints.values().forEach(endpointList -> dbEndpoints.addAll(endpointList));
```

Comment:
Do you want to pass all of the DB endpoints to the store or just the one(s) in local colo?  If local endpoint is down, the MySqlDataAccessor will try to publish the data to a remote endpoint.  My understanding from our discussion was that you didn't want that.

Reply:
Updated, to only use the db endpoints from local dc.